### PR TITLE
feat: trivial parameterization anti-patterns analysis support

### LINKED ISSUE #35: Analyzer: parameterization anti-patterns
examples:

- interpolated strings: `conn.QueryFirst<Customer>($"select * from customers where Id={id}")`
- concatenation with variables `conn.QueryFirst<Customer>("select * from customers where Id=" + id)`
- in both cases, we'll grudgingly permit constant `string` values; I'd *like* to permit other constant values, but culture rules make that impossible

(applies in all SQL scenarios, not just `QueryFirst`)

should probably be a warning, category "Security", something like "Data values should not be concatenated into SQL - use parameters instead"; it is hard to offer an auto rewrite here because the specific parameter syntax is provider specific - but we *might* know the provider (we have some stuff for that already; that's a distant second to spotting it, though)

### PR DESCRIPTION
**Pull request by DeagleGross**
PR adds the analysis of how `sql` parameter is passed:
1) if like an interpolated string `$"select * from customers where Id={id}"`, reports DAP241;
2) if like an interpolated raw string literal `$"""select * from customers where Id={id}"""`, reports DAP241;
3) if like an concatenated string `"select * from customers where Id=" + id`, reports DAP242;
4) also added a special case support for `string.Format` (reports DAP242 as in the case of concatenated strings);

Also supported lookup for local variables: in case `sql` argument is not passed directly, but via the local variable. _Example:_
```
int id = 1;
var sqlQuery = "select Id from Customers where Id = " + id;
_ = connection.Query<int>(sqlQuery);
```

| DiagnosticId | Title | Message |
| --- | --- | --- |
| DAP241 | Interpolated string usage | Data values should not be interpolated into SQL string - use parameters instead |
| DAP242 | Concatenated string usage | Data values should not be concatenated into SQL string - use parameters instead |

Closes #35

### CODE REVIEW

**Comment by mgravell** [comment]
Looks amazing! An additional possible test scenarios: does it work for interpolated raw string literals? i.e. 

`$$"""
select * from
customers where Id = {{id}}
"""
```

File: src/Dapper.AOT.Analyzers/CodeAnalysis/DapperAnalyzer.Diagnostics.cs [hunk 1]
```diff
@@ -89,6 +89,8 @@ public static readonly DiagnosticDescriptor
         DivideByZero = SqlWarning("DAP237", "Divide by zero", "Division by zero detected"),
         TrivialOperand = SqlWarning("DAP238", "Trivial operand", "Operand makes this calculation trivial; it can be simplified"),
         InvalidNullExpression = SqlWarning("DAP239", "Invalid null expression", "Operation requires a non-null operand"),
-        VariableParameterConflict = SqlError("DAP240", "Parameter/variable conflict", "The declaration of variable '{0}' conflicts with a parameter");
+        VariableParameterConflict = SqlError("DAP240", "Parameter/variable conflict", "The declaration of variable '{0}' conflicts with a parameter"),
```
Comment (mgravell) [inline src/Dapper.AOT.Analyzers/CodeAnalysis/DapperAnalyzer.Diagnostics.cs:92]: can we add /docs/rules entries for these?

File: src/Dapper.AOT.Analyzers/Internal/Inspection.cs [hunk 2]
```diff
@@ -2,6 +2,8 @@
 using Dapper.Internal.Roslyn;
 using Dapper.SqlAnalysis;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
```
Comment (mgravell) [inline src/Dapper.AOT.Analyzers/Internal/Inspection.cs:5]: see https://github.com/DapperLib/DapperAOT/tree/main/src/Dapper.AOT.Analyzers/Internal/Roslyn - I've moved all the language-specific stuff *out* - the analyzers now support both VB and C#, which needed an extra layer of abstraction. No code outside of `LanguageHelpers` should ref `.CSharp`

File: test/Dapper.AOT.Test/Verifiers/DAP241.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
+﻿using Dapper.CodeAnalysis;
+using System.Threading.Tasks;
+using Xunit;
+using Diagnostics = Dapper.CodeAnalysis.DapperAnalyzer.Diagnostics;
+namespace Dapper.AOT.Test.Verifiers;
+
+public class DAP241 : Verifier<DapperAnalyzer>
```
Comment (mgravell) [inline test/Dapper.AOT.Test/Verifiers/DAP241.cs:7]: IMO this feature warrants VB tests/support too; there should be a `VBVerifyAsync`

File: src/Dapper.AOT.Analyzers/CodeAnalysis/DapperAnalyzer.Diagnostics.cs (hunk 1 above)
Comment (DeagleGross) [reply src/Dapper.AOT.Analyzers/CodeAnalysis/DapperAnalyzer.Diagnostics.cs:92]: done

File: src/Dapper.AOT.Analyzers/Internal/Inspection.cs (hunk 2 above)
Comment (DeagleGross) [reply src/Dapper.AOT.Analyzers/Internal/Inspection.cs]: refactored to not have language-specific stuff in the `Inspection`

File: test/Dapper.AOT.Test/Verifiers/DAP241.cs (hunk 3 above)
Comment (DeagleGross) [reply test/Dapper.AOT.Test/Verifiers/DAP241.cs]: supported VB!

**Comment by DeagleGross** [comment]
@mgravell addressed PR comments

**Comment by mgravell** [comment]
Many thanks.

**Pull request merged**